hudi-integ-test/src/test/java/org/apache/hudi/integ/ITTestHoodieDemo.java

```diff
@@ -267,23 +270,37 @@ private void testSparkSQLAfterSecondBatch() throws Exception {
     assertStdOutContains(stdOutErrPair, "|GOOG  |2018-08-31 10:29:00|3391  |1230.1899|1230.085|");
   }
 
-  private void testIncrementalHiveQuery() throws Exception {
+  private void testIncrementalHiveQuery(String minCommitTimeScript, String incrementalCommandsFile,
+                                        String expectedOutput, int expectedTimes) throws Exception {
     String minCommitTime =
-        executeCommandStringInDocker(ADHOC_2_CONTAINER, MIN_COMMIT_TIME_SCRIPT, true).getStdout().toString();
+        executeCommandStringInDocker(ADHOC_2_CONTAINER, minCommitTimeScript, true).getStdout().toString();
     Pair<String, String> stdOutErrPair =
-        executeHiveCommandFile(HIVE_INCREMENTAL_COMMANDS, "min.commit.time=" + minCommitTime + "`");
-    assertStdOutContains(stdOutErrPair, "| GOOG    | 2018-08-31 10:59:00  | 9021    | 1227.1993  | 1227.215  |");
+        executeHiveCommandFile(incrementalCommandsFile, "min.commit.time=" + minCommitTime + "`");
+    assertStdOutContains(stdOutErrPair, expectedOutput, expectedTimes);
+  }
+
+  private void testIncrementalHiveQueryBeforeCompaction() throws Exception {
+    String expectedOutputCOW = "| GOOG    | 2018-08-31 10:59:00  | 9021    | 1227.1993  | 1227.215  |";
```

Comment:
minor nit: please consider renaming the variable to `expectedOutput` instead since its used across all types of tables.